### PR TITLE
migration from cassandra to postgres

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java
Comment (ChamseddineBhd) [inline src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java:52]: targetClass = String.class, is it necessary?

File: src/main/resources/application.yaml
Comment (ChamseddineBhd) [inline src/main/resources/application.yaml:15]: should be removed as the default value is false

**Review by ChamseddineBhd — APPROVED** [review]

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java:52]: Removed

File: src/main/resources/application.yaml
Comment (FranckLecuyer) [reply src/main/resources/application.yaml]: Removed

**Review by niconoir — APPROVED** [review]

**Comment by ChamseddineBhd** [comment]
You have one code smell to be fixed. (inject the field directly in the method parameter)
Add some tests to reach 80% code coverage

**Comment by FranckLecuyer** [comment]
> You have one code smell to be fixed. (inject the field directly in the method parameter)
> Add some tests to reach 80% code coverage

Code smell concerning injection of environment field has been resolved
Coverage missing concerns code for postgresql driver in DataSourceConfig class, and this code is not used during tests ...

File: src/main/java/org/gridsuite/actions/server/DataSourceConfig.java
Comment (ChamseddineBhd) [inline src/main/java/org/gridsuite/actions/server/DataSourceConfig.java:35]: database value shouldn't be in the database.properties because this file will be shared by multiple webservices (in deployment)
create a  class attribute to hold the value :     private static final String DATABASE_NAME="actions";

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/actions/server/DataSourceConfig.java
Comment (ChamseddineBhd) [reply src/main/java/org/gridsuite/actions/server/DataSourceConfig.java:35]: @niconoir  i think that this remark concern the directory server too

Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/actions/server/DataSourceConfig.java:35]: It is not the job of the env.getRequiredProperty method to look a property in the multiple property sources defined by the @PropertySource annotations, and then retrieve the 'database' property defined in the project database.properties file, because it is not defined in the database.properties file in deployment ?

**Review by ChamseddineBhd — APPROVED** [review]

File: src/main/java/org/gridsuite/actions/server/DataSourceConfig.java
Comment (jonenst) [inline src/main/java/org/gridsuite/actions/server/DataSourceConfig.java:27]: to update after recent study-server postgres changes. Maybe we should wait until it is finally merged.

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java
Comment (jonenst) [inline src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java:51]: Never use EAGER

File: src/main/resources/actions.sql
Comment (jonenst) [inline src/main/resources/actions.sql:1]: did you create this with the
```
spring.jpa.properties.javax.persistence.schema-generation.scripts.action=create
spring.jpa.properties.javax.persistence.schema-generation.scripts.create-target=create.sql
spring.jpa.properties.javax.persistence.schema-generation.scripts.create-source=metadata
```
(or equivalent) ?

File: src/main/resources/application.yaml
Comment (jonenst) [inline src/main/resources/application.yaml:15]: to move like in the latest study-server pr

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java
Comment (jonenst) [inline src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java:482]: useless because it's already in the setup

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java
Comment (jonenst) [inline src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java:378]: useless because it's already in the setup

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java
Comment (jonenst) [inline src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java:518]: useless because it's already in the setup

File: src/test/resources/application.yml
Comment (jonenst) [inline src/test/resources/application.yml:4]: to be moved to a application-default.yml file with only the overriden keys like in the recent study-server pr

File: src/main/resources/actions.sql
Comment (FranckLecuyer) [reply src/main/resources/actions.sql:1]: I created the schema with property 'spring.jpa.hibernate.ddl-auto' set to 'create' in application.yaml file

File: src/main/java/org/gridsuite/actions/server/DataSourceConfig.java
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/actions/server/DataSourceConfig.java]: Done

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java]: Done

File: src/main/resources/application.yaml
Comment (FranckLecuyer) [reply src/main/resources/application.yaml]: Done

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java]: Done

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java]: Done

File: src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java]: Done

File: src/test/resources/application.yml
Comment (FranckLecuyer) [reply src/test/resources/application.yml]: Done

File: src/test/resources/database.properties
Comment (jonenst) [inline src/test/resources/database.properties:1]: not needed ?

Comment (FranckLecuyer) [reply src/test/resources/database.properties]: file removed

**Review by JacquesBorsenberger — APPROVED** [review]

**Review by jonenst — APPROVED** [review]

**Pull request merged**
